pkg/kn/flags/bool.go

```diff
@@ -65,6 +68,25 @@ func ReconcileBoolFlags(f *pflag.FlagSet) error {
 		if err != nil {
 			return
 		}
+
+		// handle DEPRECATD flags
+		if strings.HasPrefix(flag.Usage, deprecatedPrefix) {
```

Comment:
Not sure if this is needed as you below check for "async" only.  I would wait for a second deprecated option before starting to abstract this.